### ISSUE TITLE
yadm encrypt errors

### ISSUE THREAD

**Issue by bashfulrobot** [TheLocehiliosan/yadm#85]
I suspect I must be missing something, but upon running `yadm encrypt`, I get:

```
Encrypting the following files:
/bin/ls: cannot access '.ssh/crls.d:': No such file or directory
/bin/ls: cannot access 'DIR.txt': No such file or directory
/bin/ls: cannot access '.ssh/private-keys-v1.d:': No such file or directory
/bin/ls: cannot access 'some.key': No such file or directory
/bin/ls: cannot access 'another.key': No such file or directory
/bin/ls: cannot access 'third.key': No such file or directory
```

and 

```
tar: .ssh/crls.d\:: Cannot stat: No such file or directory
tar: DIR.txt: Cannot stat: No such file or directory
tar: .ssh/private-keys-v1.d\:: Cannot stat: No such file or directory
tar: some.key: Cannot stat: No such file or directory
tar: another.key: Cannot stat: No such file or directory
tar: third.key: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
Wrote new file: /home/dustin/.yadm/files.gpg
```

Have you run into this before?

**Comment by TheLocehiliosan** [comment]
I'm not certain what's going on without a better understanding of what is in your `.yadm/encrypt` and what files and permissions exist for the data referenced in `.yadm/encrypt`.

However, some of this output makes me question if there is an issue with there being directories matched in the wildcards. There is a change in the `dev` branch which make actually address this very issue. Can you tell me if you experience the same problem using the version in the `dev` branch?

**Comment by bashfulrobot** [comment]
In my encrypt file, I simply have:

`.ssh/*`

Which I was pretty sure I have used before. 

I'll remove my installed version (installed via Ubuntu packaging tools), and try that one. I'm just commuting into work, and will give it a try when I'm there.

**Comment by TheLocehiliosan** [comment]
@bashfulrobot Any update for this? Re-reading it, I'm curious if your repo does not have the correct worktree set (you probably want it set to your `$HOME` dir). If you're still having a problem, it would be interesting to know the output of:

    yadm gitconfig core.worktree

**Comment by bashfulrobot** [comment]
Hi there, I'm just setting up my system at the moment with a rebuild... I will test this again within the day or two.

**Comment by abathur** [comment]
I'm seeing similar behavior on 1.07 (running on nixos). I think I've been able to successfully get all of my files to encrypt by explicitly adding an entry for each in my encrypt file (my other system's running a newer version of yadm).

After seeing it both throw errors and successfully encrypt files, I wondered if there's some additional quirk with the globbing in these older versions that results in a mismatch in files/paths are getting passed to yadm's subcommands? The errors might make sense if (ls, tar) are seeing either file entries or bad relative paths  that aren't being passed to others (gpg, I imagine).

I'm not sure if latest version would fix this on the nixos system, but this might make a good FAQ addition if it can be pinned down. I was quite perplexed for a while before I found the changelog note that clued me in to the fact that the version nix installs doesn't support directory expansion in `encrypt`.

Cheers!

**Comment by TheLocehiliosan** [comment]
@abathur I would be curious if using 1.12 resolves your problems. I'll have to put a test system together for nixos to test.

**Comment by abathur** [comment]
I'm curious as well. It looks like the yadm package has been updated in master to 1.12, but I'm not sure what to expect in terms of how long an update takes to hit the unstable and then stable channels.

I'm in the process of getting my feet under me with Nixos*, so I'll be sticking with their stable channel for a bit I think. It looks like they produce .vbox and .ami for each release, so it might not be too bad to set up a test system. Also possible that @magnap (who authored the update commit) or some of the other nixos users I've seen crop up in your issues may be set up to install and test from git.

*As an aside, I just successfully used yadm to preserve my local nixos config changes, wipe the system to make some big disk/partition changes, quickly restore the config, and rebuild atop them. Thanks for your work.

**Comment by abathur** [comment]
In the light of day I realize I could be a little more explicit about what I mean by seeing the same error, but getting encrypted files. I don't have access to the output I saw last night since I wiped the system shortly after seeing it. I saw the same blocks of errors @bashfulrobot reported, but between them I could see that the files encrypted. 

I want to say that this was just a list of bare file paths that I just trusted once I could see that encrypted `files` was about the right size, but I was up late enough that such claims should be treated skeptically. ;)
